src/payment/strategies/googlepay/googlepay-payment-strategy.ts

```diff
@@ -0,0 +1,210 @@
+        }
+
+        if (!this._paymentMethod || !this._paymentMethod.initializationData || !this._paymentMethod.initializationData.nonce) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        const { nonce } = this._paymentMethod.initializationData;
+
+        return this._store.dispatch(this._orderActionCreator.submitOrder(order, options))
+            .then(() =>
+                this._store.dispatch(this._paymentActionCreator.submitPayment({ ...payment, paymentData: { nonce } }))
+            )
+            .catch((error: Error) => this._handleError(error));
+    }
+
+    deinitialize(options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        return this._googlePayPaymentProcessor.teardown()
+            .then(() => super.deinitialize(options));
+    }
+
+    private _braintreeGooglePayInitializer(paymentsClient: GooglePayClient, googlePaymentInstance: GooglePayBraintreeSDK, button: Element, googlePayOptions: BraintreeGooglePayPaymentInitializeOptions): void {
+        paymentsClient.isReadyToPay(
+            {
+                allowedPaymentMethods: googlePaymentInstance.createPaymentDataRequest().allowedPaymentMethods,
+            }).then( (response: GooglePayIsReadyToPayResponse) => {
+            if (response.result) {
+                if (button) {
+                    button.addEventListener('click', (event: any) => {
+                        event.preventDefault();
+
+                        const paymentDataRequest: GooglePayBraintreePaymentDataRequest = googlePaymentInstance.createPaymentDataRequest(this._getGooglePayPaymentRequest()) as GooglePayBraintreePaymentDataRequest;
+
+                        paymentsClient.loadPaymentData(paymentDataRequest)
+                            .then((paymentData: GooglePaymentData) => {
+                                return this._setExternalCheckoutData(paymentData, googlePaymentInstance, googlePayOptions);
+                            }).catch((err: GooglePaymentsError) => {
+                                this._handleError(new Error(err.statusCode));
+                            });
+                    });
+                } else {
+                    this._handleError(new Error('GooglePay button is not ready'));
+                }
+            } else {
+                this._handleError(new Error('GooglePay is not ready to pay'));
+            }
+        });
+    }
+
+    private _getGooglePayPaymentRequest(): GooglePayPaymentDataRequest {
+        const state = this._store.getState();
+        const checkout = state.checkout.getCheckout();
+
+        if (!checkout) {
+            throw new MissingDataError(MissingDataErrorType.MissingCheckout);
+        }
+        const googlePaymentDataRequest: GooglePayPaymentDataRequest = {
+            merchantInfo: {
+                merchantId: 'your-merchant-id-from-google',
```

Comment:
How are we planning to set this `Merchant Id`? Is it going to be a value in the Merchant´s config?